envoy/upstream/outlier_detection.h

```diff
@@ -227,11 +230,11 @@ class EventLogger {
    * @param detector supplies the detector that is doing the ejection.
    * @param type supplies the type of the event.
    * @param enforced is true if the ejection took place; false, if only logging took place.
-   * @param monitor_name is optional name of failed extension monitor.
+   * @param failed_monitor is optional pointer to the failed extension monitor.
    */
   virtual void logEject(const HostDescriptionConstSharedPtr& host, Detector& detector,
                         envoy::data::cluster::v3::OutlierEjectionType type, bool enforced,
-                        absl::optional<std::string> monitor_name) PURE;
+                        absl::optional<const ExtMonitor*> failed_monitor) PURE;
```

Comment:
It's weird to use an optional raw pointer. May use `const ExtMonitor*` directly Or `OptRef<constExtMonitor>`? The nullptr could be used as nullopt anyway?

Reply:
Yeah, I just changed that type inside `absl::optional`, but you are right that raw pointer can be passed and checked for nullptr. Corrected.